server/tserver/src/main/java/org/apache/accumulo/tserver/TabletClientHandler.java

```diff
@@ -1202,33 +1202,25 @@ public void flush(TInfo tinfo, TCredentials credentials, String lock, String tab
       throw new RuntimeException(e);
     }
 
-    ArrayList<Tablet> tabletsToFlush = new ArrayList<>();
-
     KeyExtent ke = new KeyExtent(TableId.of(tableId), ByteBufferUtil.toText(endRow),
         ByteBufferUtil.toText(startRow));
 
-    for (Tablet tablet : server.getOnlineTablets().values()) {
-      if (ke.overlaps(tablet.getExtent())) {
-        tabletsToFlush.add(tablet);
-      }
-    }
-
-    Long flushID = null;
+    final Long[] flushID = {null};
```

Comment:
I think this workaround to make the object final could give you a ConcurrentModificationException or some type of concurrency problems. They made it a compile time error for a reason, this answer here gives a good explanation https://stackoverflow.com/questions/34865383/variable-used-in-lambda-expression-should-be-final-or-effectively-final

I am not sure how the equivalent logic could be accomplished with a stream. It may be better to use a traditional for loop.

Reply:
Good catch. Should be fixed by [09986b2](https://github.com/apache/accumulo/pull/2811/commits/09986b2f730c2e5cd0667de5bb01a0f04615955b).